tools/Customization/DevHome.Customization/ViewModels/FileExplorerViewModel.cs

```diff
@@ -185,37 +186,40 @@ public void RemoveTrackedRepositoryFromDevHome(string rootPath)
         RefreshTrackedRepositories();
     }
 
-    public async void AssignSourceControlProviderToRepository(string extensionName, string rootPath)
+    public async Task<SourceControlValidationResult> AssignSourceControlProviderToRepository(string extensionName, string rootPath)
     {
-        await Task.Run(async () =>
+        var result = await Task.Run(async () =>
         {
             var sourceControlExtensions = await ExtensionService.GetInstalledExtensionsAsync(ProviderType.LocalRepository);
             var extensionCLSID = sourceControlExtensions.FirstOrDefault(extension => extension.ExtensionDisplayName == extensionName)?.ExtensionClassId ?? string.Empty;
-            var result = SourceControlIntegration.ValidateSourceControlExtension(extensionCLSID, rootPath);
-            if (result.Result == ResultType.Failure)
+            var validationResult = SourceControlIntegration.ValidateSourceControlExtension(extensionCLSID, rootPath);
+            if (validationResult.Result == ResultType.Failure)
             {
                 _log.Error("Failed to validate source control extension");
-                return;
-            }
-
+                return validationResult;
+            }
+
             try
             {
                 var wrapperResult = ExtraFolderPropertiesWrapper.Register(rootPath, typeof(SourceControlProvider).GUID);
                 if (!wrapperResult.Succeeded)
                 {
                     _log.Error(wrapperResult.ExtendedError, "Failed to register folder for source control integration");
-                    return;
+                    return new SourceControlValidationResult(ResultType.Failure, ErrorType.RegistrationWithFileExplorerFailed, null, _stringResource.GetLocalized("RegistrationErrorWithFileExplorer"), _stringResource.GetLocalized("RegistrationErrorWithFileExplorer"));
```

Comment:
ExtraFolderPropertiesWrapper has an ExtendedError property. Consider using that here.